Source/Steam/CompatTool.cpp

```diff
@@ -0,0 +1,181 @@
+  if (steam_fex_tso) {
+    Options.TSO = std::strtoull(steam_fex_tso, nullptr, 0) != 0;
+  }
+
+  const auto steam_fex_multiblock = getenv("STEAM_FEX_MULTIBLOCK");
+  if (steam_fex_multiblock) {
+    Options.Multiblock = std::strtoull(steam_fex_multiblock, nullptr, 0) != 0;
+  }
+
+  // UI overrides.
+  const auto steam_fex_compat = getenv("STEAM_COMPAT_FEX_CONFIG");
+  if (steam_fex_compat) {
+    const auto steam_fex_compat_view = std::string_view(steam_fex_compat);
+    if (steam_fex_compat_view.find("TSOEnabled:1") != steam_fex_compat_view.npos) {
+      Options.TSO = true;
+    }
+    if (steam_fex_compat_view.find("Multiblock:1") != steam_fex_compat_view.npos) {
+      Options.Multiblock = true;
+    }
+    if (steam_fex_compat_view.find("ThunksDB_GL:1") != steam_fex_compat_view.npos) {
+      Options.Thunks_GL = true;
+    }
+    if (steam_fex_compat_view.find("ThunksDB_Vulkan:1") != steam_fex_compat_view.npos) {
+      Options.Thunks_Vulkan = true;
+    }
+  }
+
+  // Create the json.
+  char Buffer[4096];
+  char* Dest {};
+  Dest = json_objOpen(Buffer, nullptr);
+  {
+    Dest = json_objOpen(Dest, "Config");
+    Dest = json_str(Dest, "TSOEnabled", Options.TSO ? "1" : "0");
+    Dest = json_str(Dest, "Multiblock", Options.Multiblock ? "1" : "0");
+    Dest = json_objClose(Dest);
+  }
+
+  {
+    Dest = json_objOpen(Dest, "ThunksDB");
+    Dest = json_str(Dest, "GL", Options.Thunks_GL ? "1" : "0");
+    Dest = json_str(Dest, "Vulkan", Options.Thunks_Vulkan ? "1" : "0");
+    Dest = json_objClose(Dest);
+  }
+
+  Dest = json_objClose(Dest);
+  json_end(Dest);
+
+  // Save the json.
+  const auto ConfigPath = FEX::Config::GetConfigDirectory(false, PortableInfo);
+  const auto ConfigLocation = ConfigPath + "app_config.json";
+  if (!FHU::Filesystem::CreateDirectories(ConfigPath)) {
+    return {};
+  }
+
+  auto File = FEXCore::File::File(ConfigLocation.c_str(),
+                                  FEXCore::File::FileModes::WRITE | FEXCore::File::FileModes::CREATE | FEXCore::File::FileModes::TRUNCATE);
+
+  if (!File.IsValid()) {
+    return {};
+  }
+
+  File.Write(Buffer, strlen(Buffer));
+  return ConfigLocation;
+}
+
+int main(int argc, const char** argv) {
+  const auto PortableInfo = FEX::ReadPortabilityInformation();
+
+  const auto TemplateConfigPath = GenerateSteamConfigTemplate(PortableInfo);
+  const auto AppConfigPath = GenerateSteamAppConfig(PortableInfo);
+
+  if (!TemplateConfigPath.empty()) {
+    setenv("FEX_APP_CONFIG_LOCATION", TemplateConfigPath.c_str(), true);
+  }
+
+  if (!AppConfigPath.empty()) {
+    setenv("FEX_APP_CONFIG", AppConfigPath.c_str(), true);
+  }
+
+  const auto FEXInterpreterPath = PortableInfo.InterpreterPath + "usr/bin/FEX";
+
+  // Due to no arguments for this application, just replace argv[0] and execve again.
+  argv[0] = FEXInterpreterPath.c_str();
+  execv(FEXInterpreterPath.c_str(), const_cast<char* const*>(argv));
+
+  perror(argv[0]);
+
+  // ENOENT
+  return 127;
```

Comment:
Ah, what I meant was something like this (untested!):


```suggestion
  execv(FEXInterpreterPath.c_str(), const_cast<char* const*>(argv));
  int saved_errno = errno;

  perror(argv[0]);

  if (saved_errno == ENOENT)
    return 127;

  return 126;
```